src/main/java/de/learny/controller/TestController.java

```diff
@@ -79,24 +84,36 @@ Test update(@PathVariable("id") long id, @RequestBody Test updateTest){
 	}
 	
 	@RequestMapping(value = "/{id}/results", method = RequestMethod.GET)
-	void getResultsFromTest(@PathVariable("id") long id){
-		//TODO: Muss noch implemtiert werden
+	Iterable<TestScore> getResultsFromTest(@PathVariable("id") long id){
+		Test test = testRepository.findById(id);
+		if (test == null)
+			throw new ResourceNotFoundException("Ein Fach mit dieser id existiert nicht");
+		return test.getTestScores();
 	}
 	
 	@RequestMapping(value = "/{id}/results", method=RequestMethod.POST, consumes={MediaType.APPLICATION_JSON_VALUE})
-	void turnTest(@PathVariable("id") long id, @RequestBody Test test){
+	void turnTest(@PathVariable("id") long id, @RequestBody Test turnTest){
+		Test test = testRepository.findById(id);
+		if (test == null)
+			throw new ResourceNotFoundException("Ein Fach mit dieser id existiert nicht");
+		Account loggedInAccount = userToAccountService.getLoggedInAccount();
+		TestScore score = new TestScore(test, loggedInAccount, turnTest);
+		testScoreRepo.save(score);
```

Comment:
In TestScore.java muss die setTest methode noch angepasst werden. Außerdem sollte in Test.java die getTestscores Methode auch ein unveränderliches Set zurückgeben.